Orleans/src/Microsoft.Orleans.Host.Abstractions/StorageProvidersFeature.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the (optional) storage providers DLL.
+    public interface IStorageProvidersBuilder
+    {
+        ISiloBuilder Silo { get; }
```

Comment:
not sure why `IStorageProviderBuilder `should have a member of `ISiloBuilder`. Feels to me `IStorageProviderBuilder `should be member of `ISiloBuilder`, so it will build `StorageProvider`s based on silo's lifecycle event. Not the other way around.